src/nix/flake.cc

```diff
@@ -1343,6 +1343,39 @@ struct CmdFlakePrefetch : FlakeCommand, MixJSON
     }
 };
 
+struct CmdFlakeRoot : FlakeCommand
+{
+    bool asRef = true;
+    CmdFlakeRoot()
+    {
+        addFlag({
+            .longName = "as-ref",
+            .shortName = 'r',
+            .description = "Show flake root as a normalized flake ref.",
+            .handler = {&asRef, false}
+        });
+    }
+
+    std::string description() override
+    {
+        return "get flake root";
+    }
+
+    void run(nix::ref<nix::Store> store) override
+    {
+        std::string rootRef = getFlakeRef().to_string();
+        if (asRef) {
+            int slashIndex = rootRef.find('/');
+            while (rootRef[slashIndex + 1] == '/') {
+                slashIndex++;
+            }
+            rootRef = rootRef.substr(slashIndex);
```

Comment:
Not clear to me what this does...

Reply:
it just eats out that flake specific part